File: 926. Flip String to Monotone Increasing/bit.py
class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:
        res, one = 0, 0
        for i in s:
            if i == '1':
                one += 1
            elif one:
                one -= 1
                res += 1
        return res
# Same as Python 2

File: 926. Flip String to Monotone Increasing/bit.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int one = 0, res = 0;
        for (int index = 0; index < s.length(); index++){
            if (s[index] == '1')
                one += 1;
            else if (s[index] == '0' && one > 0) {
                one -= 1;
                res += 1;
            }
        }
        return res;
    }
};